pkg/sign/timestamping.go

```diff
@@ -41,23 +37,35 @@ type TimestampAuthorityOptions struct {
 	Retries uint
 	// Optional version string for user agent
 	LibraryVersion string
-	// Optional client (for dependency injection)
-	Client TSAClient
+	// Optional Transport (for dependency injection)
+	Transport http.RoundTripper
 }
 
 type TimestampAuthority struct {
 	options *TimestampAuthorityOptions
+	client  *http.Client
 }
 
 func NewTimestampAuthority(opts *TimestampAuthorityOptions) *TimestampAuthority {
-	return &TimestampAuthority{options: opts}
+	ta := &TimestampAuthority{options: opts}
+	ta.client = &http.Client{
+		Transport: opts.Transport,
+	}
+
+	if opts.Timeout >= 0 {
+		if opts.Timeout == 0 {
+			opts.Timeout = 30 * time.Second
+		}
+		ta.client.Timeout = opts.Timeout
+	}
+
+	return ta
 }
 
 func (ta *TimestampAuthority) GetTimestamp(ctx context.Context, signature []byte) ([]byte, error) {
 	signatureHash := sha256.Sum256(signature)
 
 	req := &timestamp.Request{
-		Certificates:  true,
```

Comment:
I went down quite the rabbit hole investigating this, but following sigstore-js' example I don't think we want to include the leaf certificate from the timestamp authority in the bundle.

Reply:
> What I don't recall is if setting this to false affects other values.

According to the RFC it should _only_ affect the inclusion of the certificate (which may contain more than one certificate) in the response.

Reply:
Perfect, that syncs up with what's in code as well.